File: gc.R
lines <- readLines("sequence.fasta")
cat("FASTA file successfully read\n")
cat("Number of lines:", length(lines), "\n")
# Remove FASTA header and blank lines
sequence <- paste(
  lines[!grepl("^>", lines) & nzchar(trimws(lines))],
  collapse = ""
)

# Convert sequence to uppercase
sequence <- toupper(sequence)

# Count nucleotides
A <- nchar(gsub("[^A]", "", sequence))
T <- nchar(gsub("[^T]", "", sequence))
G <- nchar(gsub("[^G]", "", sequence))
C <- nchar(gsub("[^C]", "", sequence))

# Calculate GC content
GC_content <- ((G + C) / nchar(sequence)) * 100

# Display results
cat("Sequence length =", nchar(sequence), "\n")
cat("A =", A, "\n")
cat("T =", T, "\n")
cat("G =", G, "\n")
cat("C =", C, "\n")
cat("GC content =", round(GC_content, 2), "%\n")

File: gc.py
# Program to calculate GC content from a FASTA file

# Function to calculate GC content
def gc_content(sequence):
    """
    Calculate the percentage of Guanine (G) and Cytosine (C)
    nucleotides in a DNA sequence.
    """
    
    # Count the number of G and C nucleotides
    g_count = sequence.count("G")
    c_count = sequence.count("C")
    
    # Calculate GC percentage
    gc_percentage = ((g_count + c_count) / len(sequence)) * 100
    
    return gc_percentage


# Open and read the FASTA file
with open("sequence.fasta", "r") as fasta_file:
    
    # Extract sequence lines and ignore the FASTA header
    sequence = "".join(
        line.strip()
        for line in fasta_file
        if not line.startswith(">")
    )


# Convert sequence to uppercase
sequence = sequence.upper()


# Calculate GC content
result = gc_content(sequence)


# Display results
print("Sequence length:", len(sequence))
print("GC Content:", round(result, 2), "%")